ReSharper.FSharp/src/FSharp.Psi/src/Impl/FSharpArgumentsOwnerUtil.cs

```diff
@@ -13,7 +13,23 @@ public static class FSharpArgumentsOwnerUtil
     public static IList<IArgument> CalculateParameterArguments(this IFSharpReferenceOwner referenceOwner,
       IEnumerable<IFSharpExpression> appliedExpressions)
     {
-      if (!(referenceOwner.Reference.GetFcsSymbol() is FSharpMemberOrFunctionOrValue mfv))
+      var fcsSymbol = referenceOwner.Reference.GetFcsSymbol();
+
+      if (fcsSymbol is FSharpUnionCase unionCase)
+      {
+        if (!unionCase.HasFields) return EmptyArray<IArgument>.Instance;
+
+        if (appliedExpressions.FirstOrDefault() is not { } appliedArg)
+          return EmptyArray<IArgument>.Instance;
+
+        appliedArg = appliedArg.IgnoreInnerParens();
+
+        return appliedArg is not ITupleExpr tupleExpr
+          ? new List<IArgument>(1) { appliedArg as IArgument }
```

Comment:
Does this `1` make any change?

Reply:
Default capacity is 4

Reply:
I know. 🙂 Does it make any measurable difference?

Reply:
Without explicit `capacity` argument, default List constructor creates an empty array. Then the initializer (which turns into list.Add) expands the list to the default capacity (4)

Reply:
Also, we have about 63 `new List<*>(1) ` code usages in the platform